src/tests/ftest/pool/evict_test.py

```diff
@@ -20,128 +20,258 @@
+        # Check that the pool was created
+        status = pool.check_files(hostlist)
+        if not status:
+            self.log.error("Invalid pool - pool data not detected on servers")
+            return None
+        # Connect to the pool
+        status = pool.connect(1)
+        if not status:
+            self.log.error("Pool connect failed or already connected")
+            return None
+        # Return connected pool
+        return pool
+
+    def evict_badparam(self, test_param):
+        """Connect to pool, connect and try to evict with a bad param.
+
+        Args:
+            test_param (str): either invalid UUID or bad server name
+
+        Returns:
+            TestPool (bool)
 
-        :avocado: tags=pool,poolevict,quick
         """
-        size = self.params.get("size", '/run/tests/sizes/size1gb/*')
-        setid = self.params.get("setname", '/run/tests/setnames/validsetname/*')
-        connectperm = self.params.get("perms",
-                                      '/run/tests/connectperms/permro/*')
+        # setup pool and connect
+        self.pool = self.connected_pool(self.hostlist_servers)
+
+        self.log.info(
+            "Pool UUID: %s\n Pool handle: %s\n Server group: %s\n",
+            self.pool.uuid, self.pool.pool.handle.value, self.pool.name)
 
+        if test_param == "BAD_SERVER_NAME":
+            # Attempt to evict pool with invald server group name
+            # set the server group name directly
+            self.pool.pool.group = ctypes.create_string_buffer(test_param)
+            self.log.info(
+                "Evicting pool with invalid Server Group Name: %s", test_param)
+        elif test_param == "invalid_uuid":
+            # Attempt to evict pool with invald UUID
+            bogus_uuid = str(uuid.uuid4())
+            # set the UUID directly
+            self.pool.pool.set_uuid_str(bogus_uuid)
+            self.log.info(
+                "Evicting pool with Invalid Pool UUID:  %s",
+                self.pool.pool.get_uuid_str())
+        else:
+            self.fail("Invalid yaml parameters - check \"params\" values")
         try:
-            uid = os.geteuid()
-            gid = os.getegid()
-
-            create_connect_evict = (
-                "{0} test-evict-pool "
-                "-m {1} "
-                "-u {2} "
-                "-g {3} "
-                "-s {4} "
-                "-z {5} {6} "
-                "-l 0".format(self.daosctl, "0731", uid, gid, setid, size,
-                              connectperm))
-            process.system(create_connect_evict)
-
-        except Exception as excep:
-            print(excep)
-            print(traceback.format_exc())
-            self.fail("Expecting to pass but test has failed.\n")
-
-
-    def test_evict_bad_pool(self):
+            # call daos evict api directly
+            self.pool.pool.evict()
+        # exception is expected
+        except DaosApiError as result:
+            if "-1005" in str(result):
```

Comment:
Is there a case where you'd get a different exception, i.e. timeout, and the subsequent checks don't yield an test failure? 

```suggestion
            status = "-1005" in str(result)
            if status:
```